src/State/Hub.php

```diff
@@ -234,16 +234,23 @@ public function startTransaction(TransactionContext $context, array $customSampl
         $tracesSampler = $options->getTracesSampler();
 
         if (null === $transaction->getSampled()) {
-            $sampleRate = null !== $tracesSampler
-                ? $tracesSampler($samplingContext)
-                : $this->getSampleRate($samplingContext->getParentSampled(), $options->getTracesSampleRate());
+            if (null !== $tracesSampler) {
```

Comment:
nit: it is always easier to read if you compare a variable against a constant
```php
if ($tracesSampler !== null) {
```

Same could be applied to a line above

Reply:
I agree, but style fixer defaults Yoda style since we use the Symfony code style.

Not sure if worth changing it since that would create a significant diff.

Reply:
I personally don't like it either, bur for the sake of the current codebase, I would not change it 🙂

Reply:
TIL! Thanks both!

@cleptric, let's think about this (out of the scope of this PR). It may be worth doing this and I am quite confident there is an automated way to cover most of the cases.

Reply:
This is something I'm happy to do in our next major, v4.0.0 😄

Reply:
For experience, I can tell you that when you get used to it it's not a big deal, and many projects use such style. There is a reason we decided long ago to use Yoda style: it prevents misassignation of variables. For example, a bug could be introduced if you do this:

```php
if ($this->foo == 'bar') {}
```

While using Yoda style would prevent it entirely because you can't assign to a constant or literal value